aws/rust-runtime/aws-config/src/http_credential_provider.rs

```diff
@@ -258,22 +254,22 @@ mod test {
 
     #[tokio::test]
     async fn successful_response() {
-        let connector = TestConnection::new(vec![successful_req_resp()]);
-        let creds = provide_creds(connector.clone()).await.expect("success");
+        let http_client = StaticReplayClient::new(vec![successful_req_resp()]);
```

Comment:
There are some good suggestions on making these test APIs nicer, but they are definitely time consuming. I think since we're going to keep aws-smithy-runtime at 0.x, we should be able to improve them in the future.